programs/clearing_house/src/controller/position.rs

```diff
@@ -779,3 +801,69 @@ fn calculate_quote_asset_amount_surplus(
 
     Ok((quote_asset_amount, quote_asset_amount_surplus))
 }
+
+pub fn update_unsettled_pnl(
+    market_position: &mut MarketPosition,
+    market: &mut Market,
+    unsettled_pnl: i128,
+) -> ClearingHouseResult<i128> {
+    let new_user_unsettled_pnl = market_position
+        .unsettled_pnl
+        .checked_add(unsettled_pnl)
+        .ok_or_else(math_error!())?;
+
+    // update market state
+    if unsettled_pnl > 0 {
+        if market_position.unsettled_pnl >= 0 {
+            // increase profit
+            market.unsettled_profit = market
+                .unsettled_profit
+                .checked_add(unsettled_pnl.unsigned_abs())
+                .ok_or_else(math_error!())?;
+        } else {
+            // decrease loss
+            market.unsettled_loss = market
```

Comment:
its bucketing similar to increase, reduce, and close/flip.

that we you can know the discrete profit and loss rather than the aggregate profit and loss (pnl)